### PR TITLE
Add custom content outlet for IgxGridToolbar - 7.0.x

### PR DESCRIPTION
**Pull request by rmkrmk**
#2983

### Additional information (check all that apply):
 - [x] New functionality

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code
 - [x] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [x] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Comment by tachojelev** [comment]
**Issue:**

@rmkrmk Currently if we provide a custom template for the toolbar content without having any of the default buttons and title enabled (but showToolbar is true), the toolbar is **not** visible.

File: CHANGELOG.md [hunk 1]
```diff
@@ -72,12 +72,14 @@ All notable changes for each version of this project will be documented in this
...
 
 ## 6.2.1
+
 ### Features
 - `igxGrid`, `igxChip`: Add display density DI token to igxGrid and igxChip ([#2804](https://github.com/IgniteUI/igniteui-angular/issues/2804))
 - `igxGrid`
     - Quick filter auto close ([#2979](https://github.com/IgniteUI/igniteui-angular/issues/2979))
     - Group By: Added title to chip in Group By area ([#3035](https://github.com/IgniteUI/igniteui-angular/issues/3035))
     - Improve UX for boolean and date columns, ([#3092](https://github.com/IgniteUI/igniteui-angular/issues/3092))
+    - Added a new `igxToolbarCustomContent` directive which can be used to mark an `ng-template` which provides a custom content for the IgxGrid's toolbar ([#2983](https://github.com/IgniteUI/igniteui-angular/issues/2983))
```
Comment (kdinev) [inline CHANGELOG.md]: Just put it under `7.1.0`

File: projects/igniteui-angular/src/lib/grids/grid-base.component.ts [hunk 2]
```diff
@@ -1688,6 +1699,7 @@ export abstract class IgxGridBaseComponent extends DisplayDensityBase implements
         return this.showToolbar &&
```
Comment (rkaraivanov) [inline projects/igniteui-angular/src/lib/grids/grid-base.component.ts]: Can we simplify this Boolean expression?

Comment (kdinev) [reply projects/igniteui-angular/src/lib/grids/grid-base.component.ts]: @rmkrmk I agree with @rkaraivanov. Just show the toolbar every time `this.showToolbar === true`, even if no elements would be displayed in it.

**Pull request merged**